py/desispec/scripts/extract.py

```diff
@@ -334,7 +339,12 @@ def main_mpi(args, comm=None, timing=None):
         dw = 0.7
 
     if args.barycentric_correction :
-        barycentric_correction_factor = barycentric_correction_multiplicative_factor(img.meta)        
+        try:
```

Comment:
The same comment as above. Assuming these are the only two places this is done I think it's fine that they're copy-pasted.

Reply:
There is a ton of replicated code in `extract.main` vs. `extract.main_mpi` which is indeed a maintenance risk; issue #827 is about resolving that overall but in the meantime we continue to accrete more copied code.  Related: I'm working with @lastephey and @ziyaointl on an extraction refactor to make it easier to port to GPUs, and that's where we'll reconcile this MPI vs. non-MPI code duplication.